src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/cas1/Cas1SpaceBookingService.kt

```diff
@@ -168,6 +175,72 @@ class Cas1SpaceBookingService(
     success(result)
   }
 
+  @Transactional
+  fun recordDepartureForBooking(
+    premisesId: UUID,
+    bookingId: UUID,
+    cas1NewDeparture: Cas1NewDeparture,
+  ): CasResult<Cas1SpaceBookingEntity> = validatedCasResult {
+    val premises = premisesService.getApprovedPremises(premisesId)
+    if (premises == null) {
+      "$.premisesId" hasValidationError "doesNotExist"
+    }
+
+    val existingCas1SpaceBooking = cas1SpaceBookingRepository.findByIdOrNull(bookingId)
+    if (existingCas1SpaceBooking == null) {
+      "$.bookingId" hasValidationError "doesNotExist"
+    }
+
+    val departureReason = departureReasonRepository.findByIdOrNull(cas1NewDeparture.reasonId)
+    if (departureReason == null) {
+      "$.cas1NewDeparture.reasonId" hasValidationError "doesNotExist"
+    } else if (departureReason.serviceScope != "approved-premises") {
+      "$.cas1NewDeparture.reasonId" hasValidationError "incorrectDepartureReasonServiceScope"
+    }
```

Comment:
there is a possibility of having a service scope of '*' which would also match.

In BookingService there is a function `serviceScopeMatches`. Maybe we should make that a function on the Departure Reason Entity?

Also - we can probably simplify things by just returning 'doesNotExist' for both outcomes (logically the reason doesn't exist). This will make it easier to untangle from other CASs if we ever get there!